crates/bevy_tasks/src/task_pool.rs

```diff
@@ -451,6 +484,46 @@ mod tests {
         assert_eq!(count.load(Ordering::Relaxed), 100);
     }
 
+    #[test]
+    fn test_thread_callbacks() {
+        let counter = Arc::new(AtomicI32::new(0));
+        let start_counter = counter.clone();
+        {
+            // Build and immediately drop to terminate
+            TaskPoolBuilder::new()
+                .num_threads(10)
+                .on_thread_spawn(move || {
+                    start_counter.fetch_add(1, Ordering::Relaxed);
+                })
+                .build();
+        }
+        assert_eq!(10, counter.load(Ordering::Relaxed));
```

Comment:
would it be possible to move this assert inside the scope with a barrier?

Reply:
The intent is to test how both initialization and finalization works. I'd rather keep it outside of the scope.

Reply:
Done.